packages/events/src/router/event/index.ts

```diff
@@ -268,12 +269,26 @@ export const eventRouter = router({
         path: '/events/search'
       }
     })
+    // @todo: remove legacy scopes once all users are configured with new search scopes
     .use(requiresAnyOfScopes(CONFIG_SEARCH_ALLOWED_SCOPES))
     .input(QueryType)
+    .use(middleware.requireSearchScope)
     .output(z.array(EventIndex))
     .query(async ({ input, ctx }) => {
       const eventConfigs = await getEventConfigurations(ctx.token)
-      return getIndex(input, eventConfigs)
+      const scopes = getScopes({ Authorization: ctx.token })
+      const searchScope = findScope(scopes, 'search')
+      // Only to satisfy type checking, as findScope will return undefined if no scope is found
+      if (!searchScope) {
+        throw new TRPCError({ code: 'FORBIDDEN' })
+      }
+      const searchScopeOptions = searchScope.options
+      return getIndex(
+        input,
+        eventConfigs,
+        searchScopeOptions,
+        ctx.user.primaryOfficeId
+      )
```

Comment:
Could we instead map the `searchScopeOptions` to a clause that can be added into `input`?